### PR TITLE
Select, Mark and export selected table rows

### PR DESCRIPTION
**Pull request by deeptailor**
User's can select and mark multiple or single rows, and export the selected rows as a csv file.

NB: 

To shift select, regular click on the first row, and then shift click on the last row you want to select.

@charlesh88 - Ball is in your court now.

### CODE REVIEW

File: src/plugins/telemetryTable/components/table.vue [hunk 1]
```diff
@@ -631,8 +691,110 @@ export default {
         clearRowsAndRerender() {
             this.visibleRows = [];
             this.$nextTick().then(this.updateVisibleRows);
-        }
+        },
+        pause(pausedByButton) {
```
Comment (akhenry) [inline src/plugins/telemetryTable/components/table.vue]: Why do we care whether it was paused by the button, or by marking a row? The logic looks the same either way to me?

File: src/plugins/telemetryTable/components/table.vue [hunk 2]
```diff
@@ -631,8 +691,110 @@ export default {
...
+                    firstRowIndex = temp - 1;
+                }
+
+                for (var i = firstRowIndex + 1; i <= lastRowIndex; i++) {
+                    let row = allRows[i];
+                    row.marked = true;
+                    this.markedRows.push(row);
+                }
+
+                this.$forceUpdate();
```
Comment (akhenry) [inline src/plugins/telemetryTable/components/table.vue]: Why is this necessary?

File: src/plugins/telemetryTable/components/table.vue [hunk 3]
```diff
@@ -667,6 +829,9 @@ export default {
         this.calculateScrollbarWidth();
 
         this.table.initialize();
+
+        document.addEventListener('keydown', this.keydown);
```
Comment (akhenry) [inline src/plugins/telemetryTable/components/table.vue]: Do we need these event listeners? Could we not just use `MouseEvent.ctrlKey` and `MouseEvent.metaKey`?

File: src/plugins/telemetryTable/TelemetryTable.js [hunk 4]
```diff
@@ -47,8 +47,10 @@ define([
             this.subscriptions = {};
             this.tableComposition = undefined;
             this.telemetryObjects = [];
+            this.rowBuffer = [];
```
[the remark this answers is not in the document]
Comment (deeptailor) [reply src/plugins/telemetryTable/TelemetryTable.js]: Sorry, was left over from my last prototype. Removed

File: src/plugins/telemetryTable/TelemetryTableColumn.js [hunk 5]
```diff
@@ -56,7 +56,11 @@ define(function () {
             }
         }
 
-    };
+        isSelectable() {
+            return false;
```
[the remark this answers is not in the document]
Comment (deeptailor) [reply src/plugins/telemetryTable/TelemetryTableColumn.js]: Makes sense, I made the requested changes

File: src/plugins/telemetryTable/components/table.vue [hunk 6]
```diff
@@ -20,94 +20,131 @@
...
+            <span class="c-button__label">Export Marked Rows</span>
+        </button>
+        <button class="c-button icon-x labeled"
+                v-show="markedRows.length"
+                v-on:click="unmarkAllRows()"
+                title="Unmark All Rows">
+            <span class="c-button__label">Unmark All Rows</span>
+        </button>
+        <div class="c-separator"></div>
+        <button class="c-button icon-pause pause-play labeled"
```
[the remark this answers is not in the document]
Comment (deeptailor) [reply src/plugins/telemetryTable/components/table.vue]: Done. Table.vue accepts an optional prop (enableMarking) which will enable marking rows and show the pause/unpause button

File: src/plugins/telemetryTable/components/table.vue [hunk 7]
```diff
@@ -20,94 +20,131 @@
...
+            </table>
+        </div>
+        <!-- Content table -->
+        <div class="c-table__body-w c-telemetry-table__body-w js-telemetry-table__body-w" @scroll="scroll" :style="{ 'max-width': widthWithScroll}">
+            <div class="c-telemetry-table__scroll-forcer" :style="{ width: totalWidth + 'px' }"></div>
+            <table class="c-table__body c-telemetry-table__body js-telemetry-table__content"
+                   :style="{ height: totalHeight + 'px'}">
+                <tbody>
+                    <telemetry-table-row v-for="(row, rowIndex) in visibleRows"
+                        :key="rowIndex"
```
[the remark this answers is not in the document]
Comment (deeptailor) [reply src/plugins/telemetryTable/components/table.vue]: makes sense, coming from React, there is a tendency to use index as a key and I dont think it makes sense to be used there as well.

File: src/plugins/telemetryTable/components/table.vue (hunk 2 above)
Comment (deeptailor) [reply src/plugins/telemetryTable/components/table.vue]: sorry leftover from my last implementation, removed

File: src/plugins/telemetryTable/components/table.vue (hunk 3 above)
Comment (deeptailor) [reply src/plugins/telemetryTable/components/table.vue]: I jumped the gun here a little, just need to make sure mouseEvent.ctrlKey and mouseEvent.metaKey works in a windows environment
